Clarify Readme (#4035)
input-output-hk/cardano-wallet: README.md

## Overview

> Cardano Wallet is a software that helps you manage your Ada. You can use it to send and receive payments on the [Cardano blockchain](https://www.cardano.org).

Reviewer: Since "software" is an uncountable noun, it sounds unnatural to say "a software".

But we can say "a software application", since "applications" are countable:

```suggestion
Cardano Wallet is a software application that helps you manage your Ada. You can use it to send and receive payments on the [Cardano blockchain](https://www.cardano.org).
```

Author: I will remove the article `a`.

interface for wallets. Most users who would like to use Cardano should
start with Daedalus.

## Quickstart

> The `cardano-wallet` executable is a HTTP server which manages your wallet(s). Here is one way to start the server using Docker:

Reviewer: Hi @HeinrichApfelmus! 

I would suggest using "that" here instead of "which", since the ability to manage wallets is a defining quality of the wallet software, and not something it just happens to do (in addition to being an HTTP server). (I think you alluded to this in a separate comment!)

Additionally, would recommend changing "a HTTP server" to "an HTTP server":

```suggestion
The `cardano-wallet` executable is an HTTP server that manages your wallet(s). Here is one way to start the server using Docker:
```

Author: Ah, it looks like I mixed up `that` and `which`. Will do.

interface for wallets. Most users who would like to use Cardano should
start with Daedalus.

## Quickstart

> The `cardano-wallet` executable is a HTTP server which manages your wallet(s). Here is one way to start the server using Docker: 

Reviewer: I would propose:

```suggestion
The `cardano-wallet` application provides an HTTP server that you can use to manage your wallet(s). Here is one way to start the server using Docker: 
```

Author: I would be ok with your suggested rephrasing, but I do prefer the original one. My reasons are

* `executable` is more specific than `application`. In the context of the QuickStart, this distinction is useful — it confers that the focus of `cardano-wallet` is a single executable, not a bundle.
* `is an HTTP server` is again more specific than `provides an HTTP server` — it suggests that if you run `cardano-wallet` executable, then it will start listening on a port for HTTP requests. Again, this is useful information to be conferred in a QuickStart section, which should not only give commands to run the application, but also give a quick overview of what parts are involved — and this choice of verb subtly confers that.
* `which manages` is shorter than `that you can use to manage` without losing information. (In fact, the original suggests that `cardano-wallet` is now in charge of your wallets, not you). This shortening is a trick I picked up from [Nature's English Communication for Scientists][nature]:
    
    > A verb in the active voice does not require a person as the agent; an inanimate object is often appropriate.
    
    Put differently, making inanimate things do something, such as managing wallets, frequently improves readability.
    
    In addition, the word `which` confers that managing wallets is a defining part of `cardano-wallet`, not just an additional function `that` it can also do.

  [nature]: https://www.nature.com/scitable/ebooks/english-communication-for-scientists-14053993/126083980/

```

> wget https://raw.githubusercontent.com/cardano-foundation/cardano-wallet/master/docker-compose.yml

Reviewer: 👍🏻

NixOS users can also use the [NixOS service](https://cardano-foundation.github.io/cardano-wallet/user-guide/NixOS).

## Obtaining `cardano-wallet`

### Executables (Linux / Windows / Mac OS)

> We provide executables as part of our [releases](https://github.com/cardano-foundation/cardano-wallet/releases). Please also see the installation instructions highlighted in the release notes.

Reviewer: 👍🏻

> **Latest releases**
>
> | cardano-wallet | cardano-node (compatible versions) |
> | --- | --- |
> | `master` branch | [1.35.4](https://github.com/input-output-hk/cardano-node/releases/tag/1.35.4) |
> | [v2023-04-14](https://github.com/cardano-foundation/cardano-wallet/releases/tag/v2023-04-14) | [1.35.4](https://github.com/input-output-hk/cardano-node/releases/tag/1.35.4) |
> | [v2022-12-14](https://github.com/cardano-foundation/cardano-wallet/releases/tag/v2022-12-14) | [1.35.4](https://github.com/input-output-hk/cardano-node/releases/tag/1.35.4) |
> | [v2022-10-06](https://github.com/cardano-foundation/cardano-wallet/releases/tag/v2022-10-06) | [1.35.3](https://github.com/input-output-hk/cardano-node/releases/tag/1.35.3) |

> ### Building from source

Reviewer: 👍🏻

NixOS users can also use the [NixOS service](https://cardano-foundation.github.io/cardano-wallet/user-guide/NixOS).

## Obtaining `cardano-wallet`

### Executables (Linux / Windows / Mac OS)

> We provide executables as part of our [releases](https://github.com/input-output-hk/cardano-wallet/releases), see also the installation instructions.

Reviewer: Since the phrase following the comma is a verb phrase, I'd suggest making it a separate sentence. (Alternatively, you could use a semi-colon.)

Additionally, shouldn't we replace "input-output-hk" with "cardano-foundation" in the URL?

```suggestion
We provide executables as part of our [releases](https://github.com/cardano-foundation/cardano-wallet/releases). See the installation instructions for more details.
```

Author: I will change the link.
I will use a semicolon — the "for more details" just lengthens the sentence without adding information.

Reviewer: > I will use a semicolon — the "for more details" just lengthens the sentence without adding information.

> "We provide executables as part of our releases; see also the installation instructions."

Hmm... this still sounds a little unnatural to me.

(The point of adding "for more details" was to make it a more natural sentence, and certainly not to add useless information 😄!)

If you feel it's too long, then perhaps we can write:

> We provide executables as part of our [releases](https://github.com/cardano-foundation/cardano-wallet/releases). Please also see the [installation instructions](link).

Author: I have changed it to

> We provide executables as part of our [releases](https://github.com/cardano-foundation/cardano-wallet/releases). Please also see the installation instructions highlighted in the release notes.

as there is no unique link for the installation instructions.

> | [v2022-12-14](https://github.com/input-output-hk/cardano-wallet/releases/tag/v2022-12-14) | [1.35.4](https://github.com/input-output-hk/cardano-node/releases/tag/1.35.4) |
> | [v2022-10-06](https://github.com/input-output-hk/cardano-wallet/releases/tag/v2022-10-06) | [1.35.3](https://github.com/input-output-hk/cardano-node/releases/tag/1.35.3) |

### Building from source

See [Building](https://cardano-foundation.github.io/cardano-wallet/developers/Building)

> ### Testing

Reviewer: 👍🏻

See [Testing](https://cardano-foundation.github.io/cardano-wallet/contributing/Testing)

## History

> The `cardano-wallet` repository has been introduced for the [Shelley phase](https://roadmap.cardano.org/) of the Cardano blockchain.

Reviewer: Since the introduction of `cardano-wallet` occurred further back in time than the recent past, I'd suggest changing "has been" to "was":

```suggestion
The `cardano-wallet` repository was introduced during the [Shelley phase](https://roadmap.cardano.org/) of the Cardano blockchain.
```

Author: 👍

> Previously, during the Byron phase, the wallet was part of the [cardano-sl](https://github.com/input-output-hk/cardano-sl) repository. (This is useful to know — sometimes, the ghosts of the past haunt us in the form of obscure bugs.)

Reviewer: Would suggest to remove the `,`. When reading an em-dash, the reader will usually pause, so it feels a bit unnatural to pause again only one word later.

Additionally, I suggest that we add "come back to" before "haunt us", since "come back to haunt us" is a commonly-used idiom that feels natural in this context. 👻 

```suggestion
Previously, during the Byron phase, the wallet was part of the [cardano-sl](https://github.com/input-output-hk/cardano-sl) repository. (This is useful to know — sometimes the ghosts of the past come back to haunt us in the form of obscure bugs.)
```

Author: 👍
